### DISCUSSION TITLE
[QUESTION] What's the proper way to implement full width stretching inside group?

### DISCUSSION THREAD

**Discussion by dendrofen** [fabricjs/fabric.js#9536] [Q&A] [answered]
I'm constructing layout editor, so implementing basic object functional like:
- stretch full width
- stretch full height

It's very actual for groups, when group size is calculating according to nested objects size/position.

**Comment by ShaMan123** [comment]
I don't understand your question but you will need the new LayoutManager

**Reply by ShaMan123** [reply]
#9148

**Reply by dendrofen** [reply]
@ShaMan123 I mean full width stretching, I wanna Rect to be full width of Group
![1064645a-323e-4e7f-9c5c-c935ab97959e](https://github.com/fabricjs/fabric.js/assets/54393894/1ce00bfd-d566-48a0-9769-e04656c38ebd)

**Reply by ShaMan123** [reply]
The easiet way to do it currently is by adding a rect to the group
I plan to expose a way to do that soon
We just completed related work on the layout manager

**Reply by ShaMan123** [reply]
I am still not able to understand what you mean.
You can set the size of the rect as you wish, take in mind the plane you are working in.
```
const finalSizeInSomePlane;
const { x, y } = fabric.util.sendVectorToPlane(finalSizeInSomePlane, planeTransform, rect.calcTransformMatrix())
rect.set({ width:x , height: y })
```

**Reply by dendrofen** [reply]
@ShaMan123 Here I've created demo, implementation of full-width stretching I'm using now.

https://codesandbox.io/p/sandbox/vigilant-dirac-gn67jc

**Reply by ShaMan123** [reply, marked as answer]
It looks OK
2 things to consider

1. call group.triggerLayout() to update the group dimensions
2. using aCoords in a group requires transforming them so they are in the group plane

see https://codesandbox.io/p/sandbox/vigilant-dirac-forked-2t58ft?file=%2Findex.html%3A27%2C1-44%2C31

**Reply by vadimcoder** [reply]
Hello. Is there anything in http://fabricjs.com/articles/ about LayoutManager? I cannot see anything about it.

**Reply by asturur** [reply]
Hello @vadimcoder no there isn't.
Is one of the new things i m supposed to add to https://fabricjs.github.io that then will be moved over fabricjs.com.

**Reply by vadimcoder** [reply]
> Hello @vadimcoder no there isn't. Is one of the new things i m supposed to add to https://fabricjs.github.io that then will be moved over fabricjs.com.

Hello @asturur, oh that would be really cool.
